openquake/commonlib/source.py

```diff
@@ -83,6 +95,25 @@ class TrtModel(collections.Sequence):
     """
     POINT_SOURCE_WEIGHT = 1 / 40.
 
+    @classmethod
+    def collect(cls, sources):
+        """
+        :param sources: dictionaries with a key 'tectonicRegion'
+        :returns: an ordered list of TrtModel instances
+        """
+        source_stats_dict = {}
+        for src in sources:
+            trt = src['tectonicRegion']
+            if trt not in source_stats_dict:
+                source_stats_dict[trt] = TrtModel(trt)
+            tm = source_stats_dict[trt]
+            if not tm.sources:
+                tm.num_ruptures = 1
+                tm.sources.append(src)
```

Comment:
I have added two important comments about why it is necessary to append one source and increment num_ruptures